### PR TITLE
Alerting: Enable Alert rule `severity` tag to override VictorOps Severity setting

### PR DESCRIPTION
**Pull request by ottaviohartman**
**What this PR does / why we need it**:

Enable severity selection in VictorOps. Feature parity with PagerDuty notifier from #22736.

This gives you the ability to send events of different severity to the same service without needing a separate notification channel configured.

This change is breaking if a Grafana alert contains a `severity` key with one of `warning`, `info`, `acknowledgment`, `recovery`, or `ok`.

### CODE REVIEW

File: docs/sources/alerting/notifications.md [hunk 1]
```diff
@@ -129,6 +129,17 @@ Move any existing rules using `custom_details.myMetric` to `custom_details.queri
...
 
 > Using `dedup_key` tag will override Grafana generated `dedup_key` with a custom key.
+
+### VictorOps
+
+To set up VictorOps, provide the URL from the Grafana Integration and substitute `$routing_key` with the routing key.
+
+Setting | Description
+---------- | -----------
+Auto resolve incidents | Resolve incidents in VictorOps once the alert goes back to ok
```
Comment (oddlittlebird) [inline docs/sources/alerting/notifications.md:139]:
```suggestion
Auto resolve incidents | Resolve incidents in VictorOps once the alert goes back to ok.
```

Comment (oddlittlebird) [reply docs/sources/alerting/notifications.md:139]: Should "OK" be capitalized? Please match whatever it says in VictorOps.

File: docs/sources/alerting/notifications.md [hunk 2]
```diff
@@ -129,6 +129,17 @@ Move any existing rules using `custom_details.myMetric` to `custom_details.queri
...
+
+>**Note:** The tag `Severity` has special meaning in the [VictorOps Incident Fields](https://help.victorops.com/knowledge-base/incident-fields-glossary/). If an alert panel defines this tag key, then it is transposed to the root of the event sent to VictorOps.
```
Comment (oddlittlebird) [inline docs/sources/alerting/notifications.md:141]: Is "transposed" the right word? If something is transposed, then it switches places with something else.

Perhaps you mean appended, which is adding it to the end?

Comment (omh1280) [reply docs/sources/alerting/notifications.md:141]: copied that language from [here](https://github.com/grafana/grafana/blame/460883d0b291a5e497a07e6efd7c8ef88a507fa0/docs/sources/alerting/notifications.md#L125)

It is transposed; it replaces a field in the root of the event. Let me know if I should change this wording and the other please!

File: docs/sources/alerting/notifications.md (hunk 1 above)
Comment (omh1280) [inline docs/sources/alerting/notifications.md:139]:
```suggestion
Auto resolve incidents | Resolve incidents in VictorOps once the alert goes back to `OK`.
```

File: docs/sources/alerting/notifications.md (hunk 2 above)
Comment (oddlittlebird) [reply docs/sources/alerting/notifications.md:141]: In that case, It needs to be something like

transposed with the root

or

it replaces the root 

I recommend the second option, as simpler words are easier for not-native English speakers to understand.

Comment (omh1280) [reply docs/sources/alerting/notifications.md:141]:
```suggestion
>**Note:** The tag `Severity` has special meaning in the [VictorOps Incident Fields](https://help.victorops.com/knowledge-base/incident-fields-glossary/). If an alert panel defines this tag key, then it replaces the `message_type` in the root of the event sent to VictorOps.
```

File: docs/sources/alerting/notifications.md [hunk 3]
```diff
@@ -129,6 +129,17 @@ Move any existing rules using `custom_details.myMetric` to `custom_details.queri
...
+
+### VictorOps
+
+To set up VictorOps, provide the URL from the Grafana Integration and substitute `$routing_key` with the routing key.
+
+Setting | Description
+---------- | -----------
+Auto resolve incidents | Resolve incidents in VictorOps once the alert goes back to `OK`.
+
+>**Note:** The tag `Severity` has special meaning in the [VictorOps Incident Fields](https://help.victorops.com/knowledge-base/incident-fields-glossary/). If an alert panel defines this tag key, then it replaces the `message_type` in the root of the event sent to VictorOps.
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:141]:
```suggestion
>**Note:** The tag `Severity` has special meaning in the [VictorOps Incident Fields](https://help.victorops.com/knowledge-base/incident-fields-glossary/). If an alert panel defines this key, then it replaces the `message_type` in the root of the event sent to VictorOps.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Requested one change.

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

**Comment by jessabe** [comment]
Hej @omh1280, thanks for the contribution!

Happy to merge this but would need to have some more documentation written about the breaking changes that you outlined in your initial PR.

**Review by osg-grafana — DISMISSED** [review]
Requesting changes based on @jessabe's feedback.

File: docs/sources/alerting/notifications.md [hunk 4]
```diff
@@ -129,6 +129,17 @@ Move any existing rules using `custom_details.myMetric` to `custom_details.queri
 This behavior will become the default in a future version of Grafana.
 
 > Using `dedup_key` tag will override Grafana generated `dedup_key` with a custom key.
+
+### VictorOps
+
+To set up VictorOps, provide the URL from the Grafana Integration and substitute `$routing_key` with the routing key.
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:135]:
```suggestion
To configure VictorOps, provide the URL from the Grafana Integration and substitute `$routing_key` with a valid key.
```

File: docs/sources/alerting/notifications.md [hunk 5]
```diff
@@ -129,6 +129,17 @@ Move any existing rules using `custom_details.myMetric` to `custom_details.queri
...
+
+Setting | Description
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:137]: Do we need a table for just one item?

**Review by achatterjee-grafana — CHANGES_REQUESTED** [review]
Added copy-edit suggestions.

**Comment by achatterjee-grafana** [comment]
@omh1280, can you update the "List of supported notifiers" section with VictorOps. Thank you very much for contributing!

File: docs/sources/alerting/notifications.md (hunk 5 above)
Comment (omh1280) [reply docs/sources/alerting/notifications.md:137]: Good point, no

**Comment by omh1280** [comment]
> Hej @omh1280, thanks for the contribution!
> 
> Happy to merge this but would need to have some more documentation written about the breaking changes that you outlined in your initial PR.

Where would this documentation go in the code?

**Comment by oddlittlebird** [comment]
> Where would this documentation go in the code?

After [this section](https://github.com/grafana/grafana/blob/master/docs/sources/installation/upgrading.md#upgrading-to-v73). Please add a new section titled "Upgrading to 7.4" at the same heading level as the 7.3 section.

**Comment by oddlittlebird** [comment]
Obviously, you will need to add some context so readers of the doc know WTH you are talking about.

**Comment by omh1280** [comment]
> Obviously, you will need to add some context so readers of the doc know WTH you are talking about.

Updated that section. It's more confusing to list how it's breaking than it is to just say what the feature is.

**Comment by omh1280** [comment]
Let me know if you need anything from me

**Comment by omh1280** [comment]
@achatterjee-grafana is there anything else I should do?

**Review by achatterjee-grafana — APPROVED** [review]
LGTM

File: docs/sources/alerting/notifications.md [hunk 6]
```diff
@@ -133,6 +133,11 @@ This behavior will become the default in a future version of Grafana.
 
 > **Note:** The `state` tag overrides the current alert state inside the `custom_details` payload.
 
+### VictorOps
+
+To configure VictorOps, provide the URL from the Grafana Integration and substitute `$routing_key` with a valid key.
+
+>**Note:** The tag `Severity` has special meaning in the [VictorOps Incident Fields](https://help.victorops.com/knowledge-base/incident-fields-glossary/). If an alert panel defines this key, then it replaces the `message_type` in the root of the event sent to VictorOps.
```
Comment (oddlittlebird) [inline docs/sources/alerting/notifications.md:140]:
```suggestion
> **Note:** The tag `Severity` has special meaning in the [VictorOps Incident Fields](https://help.victorops.com/knowledge-base/incident-fields-glossary/). If an alert panel defines this key, then it replaces the `message_type` in the root of the event sent to VictorOps.
```

**Review by oddlittlebird — APPROVED** [review]

**Comment by omh1280** [comment]
Is there anything needed from me?

**Comment by achatterjee-grafana** [comment]
All set! Thank you for contributing!!

**Pull request merged**